watchtower/wtclient/fuzz_test.go

```diff
@@ -0,0 +1,114 @@
+package wtclient
+
+import (
+	"encoding/binary"
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// getAddress generates an address from the input data or returns nil if there
+// aren't enough bytes remaining in the input data.
+func getAddress(data *[]byte) net.Addr {
```

Comment:
I originally was returning the truncated data slice on every function call.  I switched to passing the pointer around because it simplified the code quite a bit.

I can easily switch if you want, but I think the current approach is cleaner on balance.